docs/team/mep.md

```diff
@@ -0,0 +1,265 @@
+MMTk Enhancement Proposal (MEP)
+===============================
+
+An MMTk Enhancement Proposal (MEP) is a more formal variant of issue.  It has a special format, and
+will undergo a more thorough review process.  Its goal is helping the MMTk developers making more
+informed decisions.
+
+MEP is inspired by the Java Enhancement Proposal, described in https://openjdk.org/jeps/1  However,
+unlike JEP which is more open-ended, MEP is more focused on making design decisions.
+
+# When is MEP required?
+
+An MEP is required when making a significant change to the design of the MMTk core.  It is
+applicable to any kind of significant changes, including but not limited to:
+
+-   Bug fixes, including performance bug fixes, that require change to a major part of the MMTk
+    core.
+-   Changes to the MMTk core to implement a feature demanded by bindings.
+-   Major refactoring to the MMTk core.
+
+One purpose of MEP is reducing the risks to the future development of MMTk.  Large-scale changes and
+public API changes usually indicate such risks, but these are only indicators, not criteria.  The
+assessment of risks is subjective, and we need to discuss in order to reach consensus.
+
+Note: JEP is also required for things that "require two or more weeks of engineering effort" and/or
+"are in high demand by developers or customers".  We don't judge whether we need MEP based on
+engineering effort or public demand.  Many PRs for MMTk require multiple weeks of work and rigorous
+testing, but most of them can be settled with regular issues and PRs.  We label priorities using
+GitHub issue tags, such as `P-normal`, `P-high`, etc.  If a feature is requested often, and we have
+man power for that, we can raise the priority.
+
+# Format
+
+A MEP will be posted as a GitHub issue in the `mmtk-core` repository.  It should contain certain
+tags:
+
+-   **The `MEP` tag**
+    -   It is used to identify MEPs.
+-   **Area (`A-*`)**
+    -   For example, `A-gc-algorithm`.
+-   **Category (`C-*`)**
+    -   For example, `C-refactoring`.
+    -   Note that not all MEP are "enhancement" in the sense of `C-enhancement`.  Some MEPs may
+        simply be intended for fixing long-standing hard-to-fix bugs by making non-trivial changes.
+-   **Goal (`G-*`)**
+    -   For example, `G-safety`.
+
+We use the format of JEP (https://openjdk.org/jeps/2) as a frame of reference, but deviate from it
+when needed.
+
+A MEP should have the following sections:
+
+-   TL;DR (summary)
+-   Goal
+-   Non-goal (optional)
+-   Success Metric
+-   Motivation
+-   Description
+    -   Impact on Performance
+    -   Impact on Software Engineering
+-   Risks
+    -   Long Term Performance Risks
+    -   Long Term Software Engineering Risks
+    -   Impact on API
+-   Testing
+-   Alternatives (optional)
+-   Risks and Assumptions (optional)
+-   Related Issues (optional)
+
+Note: Sections in JEP but not in MEP:
```

Comment:
This is about JEP, and is irrelevant for MEP.

Reply:
I have removed that.